crates/pop-cli/src/commands/call/chain.rs

```diff
@@ -411,14 +468,45 @@ impl Call {
 				.collect();
 			full_message.push_str(&format!(" --args {}", args.join(" ")));
 		}
-		full_message.push_str(&format!(" --url {} --suri {}", chain.url, self.suri));
+		full_message.push_str(&format!(" --url {}", chain.url));
+		if self.use_wallet {
+			full_message.push_str(" --use-wallet");
+		} else {
+			full_message.push_str(&format!(" --suri {}", self.suri));
+		}
 		if self.sudo {
 			full_message.push_str(" --sudo");
 		}
 		full_message
 	}
 }
 
+// Sign and submit an extrinsic using wallet integration.
+async fn submit_extrinsic_with_secure_signing(
```

Comment:
maybe "_with_wallet" is more consistent?

Reply:
You are right, makes more sense use wallet here.